lib/core/test/unit/Cardano/Wallet/Api/TypesSpec.hs

```diff
@@ -807,9 +832,25 @@ instance
             , (5, pure $ ApiMnemonicT y)
             ]
 
-instance Arbitrary ApiBlockData where
-    arbitrary = ApiBlockData <$> genUniformTime <*> arbitrary
-    shrink (ApiBlockData t b) = ApiBlockData t <$> shrink b
+instance Arbitrary ApiTimeReference where
+    arbitrary = ApiTimeReference <$> genUniformTime <*> arbitrary
+    shrink (ApiTimeReference t b) = ApiTimeReference t <$> shrink b
+
+instance Arbitrary ApiBlockReference where
+    arbitrary = ApiBlockReference <$> arbitrary <*> arbitrary
+    shrink = genericShrink
+
+instance Arbitrary ApiNetworkInformation where
+    arbitrary = genericArbitrary
+    shrink = genericShrink
+
+instance Arbitrary NtpStatus where
+    arbitrary = pure NtpUnavailable
```

Comment:
Just wondering: why is this fixed to `NtpUnavailable` when there are two other possibilities?

Reply:
Haha! I knew I should have left a comment (I'll add one). 
The swagger document says that we will always return an `Unavailable` status. So here, we can simply return any of the branches. Now initially, I had this fully generic here but, had only one branch at the type definition. Problem is that, if we don't have at least two constructors, the generic JSON instance gets wrong and only generate strings whereas we want tagged objects!